cmd/juju/charmhub/download.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmhub
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"syscall"
+
+	"github.com/juju/charm/v8"
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+
+	"github.com/juju/juju/api/modelconfig"
+	"github.com/juju/juju/charmhub"
+	"github.com/juju/juju/charmhub/transport"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/environs/config"
+)
+
+const (
+	downloadSummary = "Locates and then downloads a CharmHub charm."
+	downloadDoc     = `
+Download a charm from the CharmHub store by a specified name.
+
+Examples:
+    juju download postgresql
```

Comment:
Depending on how the resolving of QA issues is done, this example may require updating.